contracts/strategy/ConverterStrategyBase.sol

```diff
@@ -0,0 +1,332 @@
+
+  // approx one month for average block time 2 sec
+  uint private constant _LOAN_PERIOD_IN_BLOCKS = 3600 * 24 * 30 / 2; // TODO check
+
+  uint private constant LIQUIDATION_SLIPPAGE = 5_000; // 5%
+
+  // *************************************************************
+  //                        VARIABLES
+  //                Keep names and ordering!
+  //                 Add only in the bottom.
+  // *************************************************************
+
+  /// @dev Amount of underlying assets invested to the pool.
+  uint private _investedAssets;
+
+  /// @dev Linked Tetu Converter
+  ITetuConverter public tetuConverter;
+
+  /// @dev Linked Tetu Liquidator
+  ITetuLiquidator public tetuLiquidator;
+
+  bool private _isReadyToHardWork;
+
+  // *************************************************************
+  //                        INIT
+  // *************************************************************
+
+  /// @notice Initialize contract after setup it as proxy implementation
+  function __ConverterStrategyBase_init(
+    address controller_,
+    address splitter_,
+    address converter_
+  ) internal onlyInitializing {
+    __StrategyBase_init(controller_, splitter_);
+    _requireInterface(converter_, InterfaceIds.I_TETU_CONVERTER);
+    tetuConverter = ITetuConverter(converter_);
+    _isReadyToHardWork = true;
+
+  }
+
+  // *************************************************************
+  //                     RESTRICTIONS
+  // *************************************************************
+
+  /// @dev Restrict access only for splitter
+  function _onlyTetuConverter() internal view {
+    require(msg.sender == address(tetuConverter), "CSB: Only TetuConverter");
+  }
+
+  // *************************************************************
+  //                        VIEWS
+  // *************************************************************
+
+  /// @dev See {IERC165-supportsInterface}.
+  function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+    return interfaceId == InterfaceIds.I_CONVERTER_STRATEGY || super.supportsInterface(interfaceId);
+  }
+
+  // *************************************************************
+  //                       OVERRIDES StrategyBase
+  // *************************************************************
+
+  /// @dev Amount of underlying assets converted to pool assets and invested to the pool.
+  function investedAssets() override public view virtual returns (uint) {
+    return _investedAssets;
+  }
+
+  /// @dev Deposit given amount to the pool.
+  function _depositToPool(uint amount) override internal virtual {
+    _doHardWork(false);
+    if (amount == 0) return;
+
+    address _asset = asset;
+    uint assetBalanceBefore = _balance(_asset);
+
+    address[] memory tokens = _depositorPoolAssets();
+    uint len = tokens.length;
+    uint[] memory tokenAmounts = new uint[](len);
+    uint assetAmountForToken = amount / len;
```

Comment:
need to calc this amount specifically to each pool
not all pools have 50/50 ratios
call this function in adapter where it will be calc specifically